[docs] Emphasize how to avoid failing tests when migrating from v4 to v5 (#32159)
mui/material-ui: docs/data/material/guides/migration-v4/migration-v4.md

## `ThemeProvider` setup

> Before upgrading to v5, please make sure that `ThemeProvider` is defined at the root of your application, including in tests,  (even if you are using the **default theme**) and **NO** `useStyles` is called before `<ThemeProvider>`.

Reviewer: ```suggestion
Before upgrading to v5, please make sure that `ThemeProvider` is defined at the root of your application and in tests—even if you are using the default theme—and `useStyles` is _not_ called before `ThemeProvider`.
```
